### PR TITLE
chore: start collectors architecture

### PR DESCRIPTION
**Pull request by prokopsimek**
## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:

- [ ] I have updated the documentation accordingly.
- [ ] I have added new practice to practice list in README.md.
- [ ] I have read the **CONTRIBUTING** document.
- [ ] I haven't repeated the code. (DRY)
- [ ] I have added tests to cover my changes.
- [ ] All new and existing tests passed.

### CODE REVIEW

File: src/services/bitbucket/BitbucketService.ts [hunk 1]
```diff
@@ -522,9 +524,30 @@ export class BitbucketService implements IVCSService {
...
-    throw new Error('Method not implemented yet.');
+    const params: Params.RepositoriesListCommits = {
+      repo_slug: repo,
+      workspace: owner,
+    };
+    const commits = await this.paginate(this.client.repositories.listCommits, params);
+
+    return (
+      commits
+        //filter diplicate commiter names
```
Comment (adelkahomolova) [inline src/services/bitbucket/BitbucketService.ts:537]:
```suggestion
        //filter duplicate committer names
```

File: src/services/bitbucket/BitbucketService.ts [hunk 2]
```diff
@@ -560,6 +583,16 @@ export class BitbucketService implements IVCSService {
     };
   }
 
+  private async paginate(method: (params: Params.RepositoriesListCommits) => AsyncResponse<any>, params: Params.RepositoriesListCommits) {
```
Comment (adelkahomolova) [inline src/services/bitbucket/BitbucketService.ts:586]: rename or make it generic

File: src/services/gitlab/GitLabService.ts [hunk 3]
```diff
@@ -403,8 +404,32 @@ export class GitLabService implements IVCSService {
     return data;
   }
 
-  async listContributors(owner: string, repo: string): Promise<Paginated<Contributor>> {
-    throw new Error('Method not implemented yet.');
+  async listContributors(owner: string, repo: string, options?: ListGetterOptions): Promise<Contributor[]> {
+    const commits = await this.getAllCommits(`${owner}/${repo}`, options?.pagination);
+    const items = await Promise.all(
+      commits
+        //filter diplicate commiter names
```
Comment (adelkahomolova) [inline src/services/gitlab/GitLabService.ts:411]:
```suggestion
        //filter duplicate committer names
```

**Review by adelkahomolova — CHANGES_REQUESTED** [review]

**Review by adelkahomolova — APPROVED** [review]

**Review by Jakub-Vacek — APPROVED** [review]

**Pull request merged**